### PR TITLE
Add theme switcher in an App Bar

### PR DESCRIPTION
**Pull request by julien-deramond**

### CODE REVIEW

**Comment by julien-deramond** [comment]
Wit the `macOS (desktop)` target, run the first time perfectly.
Now I got the following errors:

```
Warning: CocoaPods not installed. Skipping pod install.
  CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
  Without CocoaPods, plugins will not work on iOS or macOS.
  For more info, see https://flutter.dev/platform-plugins
To install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

Exception: CocoaPods not installed or not in valid state.
```

Plus new files added and modified:

```
On branch main-jd-add-theme-switcher
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   ios/Flutter/Debug.xcconfig
	modified:   ios/Flutter/Release.xcconfig
	modified:   macos/Flutter/Flutter-Debug.xcconfig
	modified:   macos/Flutter/Flutter-Release.xcconfig
	modified:   macos/Flutter/GeneratedPluginRegistrant.swift

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	ios/Podfile
	macos/Podfile

no changes added to commit (use "git add" and/or "git commit -a")
```

Not sure if they must be commited/pushed.

**Comment by julien-deramond** [comment]
OK so I can run it again after:
```
sudo gem update --system 3.4.12
sudo gem install cocoapods
```

Now have this changes, not sure if they must be committed or not 🤷 
```
On branch main-jd-add-theme-switcher
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   ios/Flutter/Debug.xcconfig
	modified:   ios/Flutter/Release.xcconfig
	modified:   macos/Flutter/Flutter-Debug.xcconfig
	modified:   macos/Flutter/Flutter-Release.xcconfig
	modified:   macos/Flutter/GeneratedPluginRegistrant.swift
	modified:   macos/Runner.xcodeproj/project.pbxproj
	modified:   macos/Runner.xcworkspace/contents.xcworkspacedata

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	ios/Podfile
	macos/Podfile
	macos/Podfile.lock
```

File: app/lib/ui/main_screen.dart [hunk 1]
```diff
@@ -36,41 +40,75 @@ class _MainScreenState extends State<MainScreen> {
...
+    return Consumer<ModelTheme>(builder: (context, ModelTheme themeNotifier, child)
+    {
+      return Scaffold(
+          appBar: AppBar(
+            title: Text(_MainScreenConfig.mainMenuItems[_selectedIndex].label),
+            actions: [
+              DropdownButton<String>(
+                  value: chosenTheme,
+                  items: themeList.map<DropdownMenuItem<String>>((String value) {
+                    return DropdownMenuItem<String>(
```
Comment (ludovic35) [inline app/lib/ui/main_screen.dart:52]: put it in a dedicated widget ThemeSelector for example

**Review by ludovic35 — CHANGES_REQUESTED** [review]

File: app/lib/model_theme.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import 'package:flutter/material.dart';
+import 'theme_preference.dart';
+
+class ModelTheme extends ChangeNotifier {
+  late ThemeMode _theme;
```
[the remark this answers is not in the document]
Comment (julien-deramond) [reply app/lib/model_theme.dart]: Yep totally right! Fixed via https://github.com/Orange-OpenSource/ods-flutter/pull/72/commits/a61241cf5047c354e1e8c33216c9f8890d901aae

File: app/lib/model_theme.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  //Switching the themes
+  set theme(ThemeMode value) {
+    _theme = value;
+    _preferences.setTheme(value);
+    notifyListeners();
+  }
+
+  getPreferences() async {
```
[the remark this answers is not in the document]
Comment (julien-deramond) [reply app/lib/model_theme.dart]: Fixed in https://github.com/Orange-OpenSource/ods-flutter/pull/72/commits/96484904accfb3f849bf75c31aaeaf8603e77a44

File: app/lib/ui/main_screen.dart (hunk 1 above)
Comment (julien-deramond) [reply app/lib/ui/main_screen.dart:52]: Yep indeed way better if we do that. Let me tackle it in a separate branch + PR.

**Pull request merged**